### ISSUE TITLE
Base Metals Suite Support

### ISSUE THREAD

**Issue by Dragnier** [Resynth-Minecraft-Mod/Resynth-Mod#7]
Could you add support for Base Metals, Modern Metals, and Base Minerals that is independent of Thermal?

**Comment by LMelaia** [comment]
I'm not sure I completely understand you.

If you're suggesting adding copper, tin, ect plants for other mods that add these ores (Forestry, Industrial Craft 2 and so on). I am contemplating adding plants for these mods that just use the thermal foundation textures so that Thermal Foundation isn't required to get base metal plants. If I can't figure
out a better solution to adding "universal base metal plants" this is probably the route I'll take. It will be a while before this is implemented though, as I'd like to add support for some other mods ores and resources before I get to this.

**Comment by Dragnier** [comment]
Use the Forge Ore Dictionary to find metals instead of direct names from mods, maybe? Anything registered to oreCopper by a mod should be found that way. Though, some mods do not oredict and others make their own odd ones. Most of the big mods like Thermal, Base Metals, IE, etc. use the proper oredict entries.

**Comment by LMelaia** [comment]
As it stands, registering plants with ores/metals/ect. from the Ore Dictionary won't actually help much. This is mostly due to Resynth being written the way it is.

Each plant (along with its textures and config settings) is specifically and manually written for each ore and resource. This makes implementing a registering system for these kinds of plants from the Ore Dict rather difficult as the Ore Dict only gives a block/item reference which doesn't help in creating a plant. Each plant will still have to be manually modeled/textured and named with its own unique name - the Ore Dict won't help much with this.

My current plan of action is just to make specific plant sets for each of the mods that add one of the ores in question - what I did for Thermal Expansion (quite tedious but it does end up making reasonably tailored plants). I will definitely look into an easier way of doing it before actually implementing it but, as it currently stands, this is my best looking option.

Ultimately, the end results from either method should be the same.

---

I will keep the issue open until the plants are actually implemented into the mod and to (maybe) update on the progress of this specific issue .

**Comment by Dragnier** [comment]
All versions of any ore should use the same plant. Why would you make multiple? There should be no need for a Thermal copper plant, a Mekanism copper plant, and a Base Metals copper plant. Most pack makers will typically make sure that only one type of an ore spawns in the world anyway and the resulting dropped ore, ingot, etc. from the plant will typically be oredicted so it doesn't matter for recipes. The only plants you should have to make custom are for ores that do not have common oredict entries. For example, Amber from Thaumcraft or Biomes O' Plenty.

Basically, you are far over-complicating what you need.

Or, are you talking about how you spawn the seeds? You should be able to detect an ore broken's oredict entry and drop based on that instead of checking for specific blocks.

**Comment by LMelaia** [comment]
The two biggest issues are::

1. The checking for broken/blown up ores is designed to check each ore that is destroyed against a plant set that defines itself as a plant for the ore. This can probably be changed quite easily though so it's not my biggest concern.
2. My biggest issue is that every plants produce **needs** to smelt into a usable resource (preferably into the ore the seeds were originally obtained from). This is what makes defining a plant set for all ores difficult. If Ore Dict works to make Organic Ore act exactly as Thermal Foundation Ore, then Ore Dict plants may be more possible than I originally thought. This depends on how **exactly** Ore Dict works though.

I'm thinking of writing a custom `OreDictPlantSet` that auto defines a plant set from Ore Dict names that has its own logic to drop seeds. This should work, at least, in theory. I'll start doing some testing once I'm done with version 1.3.3.

**Comment by Dragnier** [comment]
Any progress on this in the newer versions? (1.15.2?) I'm currently using Silent's Mechanisms and curious about whether your mod will pick up on the ores.

**Comment by LMelaia** [comment]
I sincerely apologise for the incompetence on my part when dealing with this request, as well as for the wait and lack of responses. I think it's about damn time that I actually get to this and provide an explanation for what happened and the plans moving forward.

### Silent's Mechanisms ore crop support (issue #44)

Firstly, I just noticed your last message about Silent's Mechanisms - I do not know how I hadn't seen it before. However support for Silent's Mechanisms can be added in the next release for 1.16 (which should be soon). I've created separate ticket (#44) for it, which I'll close when the next version comes out.

### Base Metals ore crop support

As for Base Metals, well, I don't believe I'll be adding support for it after all. I'm sorry. I had originally intended to add support in 1.12, however I got caught up with all the changes while developing the 1.13 version. I was then planning on adding support for the mod in a later release, when it updated to a newer version of Minecraft. Although as time went on, development on the 1.12 version became difficult with all the changes and I stopped updating it, and Base Metals has yet to update to a new version (with no sign of a port in development either), and as time goes on this thread is getting ignored and forgotten, so I've decided to just call it quits at this point. Sorry again.

**Should Base Metals ever update in the future and you still want support for the ores, don't hesitate to reopen this issue.**

P.S. integration with ores from other mods has gotten much better with the changes to how mod integration and plants work in the update to 1.13 and onwards, so any future requests for modded ore integration will be easier and faster, should you want support for any other modded ores.
